netaddr.go

```diff
@@ -224,17 +169,12 @@ func ParseIP(s string) (IP, error) {
 
 	if !strings.Contains(s, ":") {
 		if ip4 := ipa.IP.To4(); ip4 != nil {
-			var v4 v4Addr
-			copy(v4[:], ip4)
-			return IP{v4}, nil
+			return IPv4(ip4[0], ip4[1], ip4[2], ip4[3]), nil
 		}
 	}
-	var v6 v6Addr
-	copy(v6[:], ipa.IP.To16())
-	if ipa.Zone != "" {
-		return IP{v6AddrZone{v6, ipa.Zone}}, nil
-	}
-	return IP{v6}, nil
+	var a16 [16]byte
+	copy(a16[:], ipa.IP.To16())
```

Comment:
It makes me sad what terrible code the compiler generates for this. Not important enough to be worth changing here (the fix is to do two binary.NativeEndian.[Put]Uint64 calls), just...sad.

Reply:
File a Go bug?